### ISSUE TITLE
Use async functions instead of promises

### ISSUE THREAD

**Issue by mrts** [jhipster/jhipster-vuejs#447]
##### **Overview of the feature request**

Use async/await instead of promises for more succinct code.

For example, instead of

```ts
    this.rideService()
      .retrieve(paginationQuery)
      .then(
        res => {
          this.rides = res.data;
          this.totalItems = Number(res.headers['x-total-count']);
          this.queryCount = this.totalItems;
          this.isFetching = false;
        },
        err => {
          this.isFetching = false;
        }
      );
```

do this:

```ts
  try {
    const res = await this.rideService().retrieve(paginationQuery);
    this.rides = res.data;
    this.totalItems = Number(res.headers['x-total-count']);
    this.queryCount = this.totalItems;
  } finally {
    this.isFetching = false;
  }
```

##### **Motivation for or Use Case**

Quoting [Google Developers async functions primer](https://developers.google.com/web/fundamentals/primers/async-functions):

> Async functions are [...] quite frankly marvelous. They allow you to write promise-based code as if it were synchronous, but without blocking the main thread. They make your asynchronous code less "clever" and more readable.

---

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by atomfrede** [comment]
Seems like a good idea. Would you mind doing a PR for that? @jhipster/vuejs-developers wdyt?

**Comment by mrts** [comment]
Possibly - let's see how far I get

**Comment by antonioortizpola** [comment]
@mrts have you found time to work on this?

If not, I could take it to work with in the next one or two weeks.

**Comment by pascalgrimaud** [comment]
I don't know Vue.js enough to tell if it's the right way or not.
Cc @sahbi-ktifa @hdurix @nonomoho : I need your opinion here
Maybe @gnuk you can help too ? What do you think

**Comment by nonomoho** [comment]
I discussed it with @Gnuk:
Both solutions are good but we don't like try/catch so much ...

**Comment by mrts** [comment]
@antonioortizpola nope, sorry, I don't currently have time. Please go ahead - and good luck :)! Note that you have to use try/catch, try/finally is incomplete.

**Comment by mrts** [comment]
@nonomoho glad you like this, but can you elaborate why you don't like try/catch?

**Comment by nonomoho** [comment]
It's juste personal, I think `then` syntax is more understandable! 
But it's just my opinion 😉

**Comment by antonioortizpola** [comment]
Coming from C# await seems way more natural to me, what is the "standard" or the more used way in JS and Vue?
